pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/PublisherStatsImpl.java

```diff
@@ -53,35 +53,13 @@ public class PublisherStatsImpl implements PublisherStats {
     public boolean supportsPartialProducer;
 
     /** Producer name. */
-    @JsonIgnore
-    private int producerNameOffset = -1;
-    @JsonIgnore
-    private int producerNameLength;
-
+    private String producerName;
```

Comment:
How can these changes affect JSON serde code path? I don't know if we pass or persist these classes via JSON somewhere.

If no, `@JsonIgnore` can be redundant. If so, adding new serialize field may affect especially this is a class under pulsar-common.

Reply:
The former `@JsonIgnore` annotations were needed because of the extra ordinary solution to store String values. They aren't needed anymore.